File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int count = 1; 
        int pos = 1; 
        for(int i = 1; i< nums.length; i++){
            if(nums[i] == nums[i-1]){
                count++; 
            } else{
                count = 1; 
            }

            if(count <= 2){
                nums[pos] = nums[i];
                pos++; 
            }
        }

        return pos; 
    }
}

File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if(nums.size() <= 2) return nums.size(); 
        int pos = 2; 
        for(int i = 2; i < nums.size(); i++){
            if(nums[i] != nums[pos-2]){
                nums[pos++] = nums[i]; 
            }
        }
        return pos; 
    }
};